python/src/scipp/compat/mantid.py

```diff
@@ -197,21 +197,127 @@ def validate_and_get_unit(unit, allow_empty=False):
         return known_units[unit]
 
 
-def init_pos(ws):
-    nHist = ws.getNumberHistograms()
-    pos = np.zeros([nHist, 3])
-
+def _to_spherical(input):
+    input["r"] = sc.sqrt(input["x"].data * input["x"].data +
+                         input["y"].data * input["y"].data +
+                         input["z"].data * input["z"].data)
```

Comment:
Any reason for not using https://scipp.github.io/generated/scipp.dot.html#scipp.dot based on the position?

Reply:
Good point.